### PR TITLE
to-html: init at 0.1.4

### PR DESCRIPTION
**Pull request by icewind1991**
###### Description of changes

Add [to-html](https://github.com/Aloso/to-html) 0.1.3, a "Terminal wrapper for rendering a terminal on a website by converting ANSI escape sequences to HTML".

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GabrielDougherty** [comment]
Result of `nixpkgs-review pr 221364` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>to-html</li>
  </ul>
</details>

**Review by GabrielDougherty — APPROVED** [review]

**Comment by icewind1991** [comment]
updated to 0.1.4

File: pkgs/tools/text/to-html/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+rustPlatform.buildRustPackage rec {
+  version = "0.1.4";
+  pname = "to-html";
+
+  src = fetchFromGitHub {
+    owner = "Aloso";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-zkTBjsMFhRz7lVRh8i+XkaJ/qWmTAMPnkH5aDhbHf8U=";
```
Comment (drupol) [inline pkgs/tools/text/to-html/default.nix:15]:
```suggestion
    hash = "sha256-zkTBjsMFhRz7lVRh8i+XkaJ/qWmTAMPnkH5aDhbHf8U=";
```

File: pkgs/tools/text/to-html/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  };
+
+  cargoSha256 = "sha256-hXc+lB3DKnRZkp1U5wW/vPKSZ0c1UknQCAxDfE7Eubg=";
```
Comment (drupol) [inline pkgs/tools/text/to-html/default.nix:18]:
```suggestion
  cargoHash = "sha256-hXc+lB3DKnRZkp1U5wW/vPKSZ0c1UknQCAxDfE7Eubg=";
```

I'm pretty sure you can use `cargoHash` here instead.

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution. I've left some comments.

**Comment by drupol** [comment]
@icewind1991 I've use [`nix-init`](https://github.com/nix-community/nix-init) (`nix run nixpkgs#nix-init -- -u https://github.com/Aloso/to-html`) to generate the nix file for this program, here it is:

<details>

```nix
{ lib
, rustPlatform
, fetchFromGitHub
}:

rustPlatform.buildRustPackage rec {
  pname = "to-html";
  version = "0.1.4";

  src = fetchFromGitHub {
    owner = "Aloso";
    repo = "to-html";
    rev = "v${version}";
    hash = "sha256-zkTBjsMFhRz7lVRh8i+XkaJ/qWmTAMPnkH5aDhbHf8U=";
  };

  cargoHash = "sha256-hXc+lB3DKnRZkp1U5wW/vPKSZ0c1UknQCAxDfE7Eubg=";

  meta = with lib; {
    description = "Utilities for making the colo documentation";
    homepage = "https://github.com/Aloso/to-html";
    changelog = "https://github.com/Aloso/to-html/blob/${src.rev}/CHANGELOG.md";
    license = licenses.mit;
    maintainers = with maintainers; [ ];
  };
}
```

</details>

Can you compare and see if you're not missing anything? I quickly notice that you could add the `meta.changelog` attribute.

**Comment by icewind1991** [comment]
added changelog

**Review by drupol — APPROVED** [review]

**Pull request merged**